### ISSUE TITLE
Messages out of order due to clock change

### ISSUE THREAD

**Issue by anshin** [signalapp/Signal-Android#3619]
I had a TextSecure (not SMS, didn't try SMS) conversation (other participant was using Signal on iOS in case that matters) this morning with my phone's clock (inadvertently) set an hour ahead of the correct time. Messages received were marked as being sent at the correct time (and were not shown using relative time such as "an hour ago") but as being received at my clock's time. When I corrected the time, the received time did not change, and new messages, marked as being received at the correct (and current) time, were placed in the thread before the older messages rather than appended to the thread.

Device info: LG Optimus Exceed 2, stock android 4.4.2, TextSecure 2.21.0

App state: Push registered, set as default SMS app

**Comment by wp9015362** [comment]
Might be related to the following issues:

https://github.com/WhisperSystems/TextSecure/issues/2385
https://github.com/WhisperSystems/TextSecure/issues/3618

?

**Comment by malenki** [comment]
Android 4.4.2
Signal 3.16.0
I experienced (and still do) a similar issue today.
With one person I exchanged messages on:
Day A, 22:00
**Day B, 14:00**
Day B, 20:30
Day B, 20:35
Day B, 20:40
Day B, 20:45

Now there was obviously a glitch in my Telco providers network which caused my to show 01:37am although it was 17:00 (5pm). Thus I set the clock manually.
Although between 14:00 and 20:30 no messages were exchanged, Signal sorts the messages this way: 
Day A, 22:00
Day B, 20:30
Day B, 20:35
Day B, 20:40
Day B, 20:45
**Day B, 14:00**

The contacts list also shows this person as "last chatted with: 6h ago"
